core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/RunQC.java

```diff
@@ -12,93 +12,242 @@
+    } else {
+      return this.getId() == them.getId();
+    }
+  }
 
-  /**
-   * Sets a flag describing if this run should be processed via some form of primary analysis
-   * 
-   * @param doNotProcess
-   *          boolean.
-   */
-  public void setDoNotProcess(boolean doNotProcess);
+  public boolean getDoNotProcess() {
+    return doNotProcess;
+  }
 
-  /**
-   * Returns a list of partitions that should be processed as part of a downstream primary analysis
-   * 
-   * @return List<Partition> partitionSelections
-   */
-  public List<Partition> getPartitionSelections();
+  public long getId() {
+    return qcId;
+  }
 
-  /**
-   * Sets a list of partitions that should be processed as part of a downstream primary analysis
-   * 
-   * @param partitionSelections
-   *          List<Partition>
-   */
-  public void setPartitionSelections(List<Partition> partitionSelections);
+  public String getInformation() {
+    return information;
+  }
+
+  public List<Partition> getPartitionSelections() {
+    return partitionSelections;
+  }
+
+  public String getQcCreator() {
+    return qcCreator;
+  }
+
+  public Date getQcDate() {
+    return qcDate;
+  }
+
+  public QcType getQcType() {
+    return qcType;
+  }
+
+  public Run getRun() {
+    return run;
+  }
+
+  @Override
+  public int hashCode() {
+    if (getId() != RunQC.UNSAVED_ID) {
+      return (int) getId();
+    } else {
+      final int PRIME = 37;
+      int hashcode = -1;
+      if (getQcCreator() != null) hashcode = PRIME * hashcode + getQcCreator().hashCode();
+      if (getQcDate() != null) hashcode = PRIME * hashcode + getQcDate().hashCode();
+      if (getQcType() != null) hashcode = PRIME * hashcode + getQcType().hashCode();
+      return hashcode;
+    }
+  }
+
+  @Override
+  public boolean isDeletable() {
+    return getId() != RunQC.UNSAVED_ID;
+  }
+
+  public void setDoNotProcess(boolean doNotProcess) {
+    this.doNotProcess = doNotProcess;
+  }
+
+  public void setId(long qcId) {
+    this.qcId = qcId;
+  }
+
+  public void setInformation(String information) {
+    this.information = LimsUtils.findHyperlinks(information);
+  }
+
+  public void setPartitionSelections(List<Partition> partitionSelections) {
+    this.partitionSelections = partitionSelections;
+  }
+
+  public void setQcCreator(String qcCreator) {
+    this.qcCreator = qcCreator;
+  }
+
+  public void setQcDate(Date qcDate) {
+    this.qcDate = qcDate;
+  }
+
+  public void setQcType(QcType qcType) {
+    this.qcType = qcType;
+  }
+
+  public void setRun(Run run) {
+    this.run = run;
+  }
+
+  @Override
+  public String toString() {
+    StringBuilder sb = new StringBuilder();
+    sb.append(getId());
+    sb.append(" : ");
+    sb.append(getQcCreator());
+    sb.append(" : ");
+    sb.append(getQcDate());
+    sb.append(" : ");
+    sb.append(getQcType());
+    return sb.toString();
+  }
+
+  public boolean userCanRead(User user) {
+    return true;
+  }
+
+  public boolean userCanWrite(User user) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "user". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)